crates/ide-completion/src/render/function.rs

```diff
@@ -153,6 +159,62 @@ fn render(
     item
 }
 
+fn compute_associated_fn(
+    ctx: &RenderContext<'_>,
+    func: hir::Function,
+    func_kind: &FuncKind<'_>,
+    db: &dyn HirDatabase,
+) -> Option<CompletionRelevanceFn> {
+    if func.as_assoc_item(ctx.db()).is_none() {
+        return None;
+    }
+
+    let has_args = !func.assoc_fn_params(db).is_empty();
+    let ret_type = func.ret_type(db);
+    let has_self_arg = func.self_param(db).is_some();
+    let ret_unit_type = ret_type.is_unit();
+    let self_type = match func_kind {
+        FuncKind::Function(PathCompletionCtx {
+            qualified: Qualified::With { path, .. }, ..
+        }) => path.segment().and_then(|ps| ps.name_ref()).map(|n| n.to_string()),
+        _ => None,
+    }; // self type without any generic args
```

Comment:
String comparing is certainly the wrong approach here, take a look at how we populate the `type_match` field of the relevance struct for how to possibly do this.

Reply:
I have pushed some commit to do this reliably.